tapchannel/aux_traffic_shaper.go

```diff
@@ -395,7 +395,14 @@ func (s *AuxTrafficShaper) ProduceHtlcExtraData(totalAmount lnwire.MilliSatoshi,
 	if htlc.Amounts.Val.Sum() > 0 {
 		log.Tracef("Already have asset amount (sum %d) in HTLC, not "+
 			"producing extra data", htlc.Amounts.Val.Sum())
-		return totalAmount, htlcCustomRecords, nil
+
+		htlc.SetNoopAdd()
+		updatedRecords, err := htlc.ToCustomRecords()
```

Comment:
Seems the aspect related to the reserve that you mentioned isn't in this PR yet?

Reply:
Ok, I see how this fits in now, resolving this comment.